Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -60,9 +62,12 @@ public RestChannel(RestServiceCollection serviceCollection, string endpoint, Cha
             var restMethod = _serviceCollection.GetRestMethod(method);
 
             var cancellationTokenSource = new CancellationTokenSource();
+            if (options.Deadline.HasValue)
+            {
+                cancellationTokenSource = new CancellationTokenSource(options.Deadline.Value - DateTime.UtcNow);
```

Comment:
Please add a TODO here to inject an IClock in the future. That doesn't need to block GA though.

Reply:
(We should probably consider what happens if the deadline has already passed, too. If we end up with a TimeSpan with TotalMilliseconds of -1, that ends up as an infinite deadline instead of one just passed. I think we can fix that shortly after GA though.)

Reply:
Not an infinite deadline, but yet another exception of a different type. Fixed.